src/hotspot/cpu/x86/c1_CodeStubs_x86.cpp

```diff
@@ -300,6 +300,22 @@ void MonitorExitStub::emit_code(LIR_Assembler* ce) {
   __ jmp(_continuation);
 }
 
+void LoadKlassStub::emit_code(LIR_Assembler* ce) {
+  __ bind(_entry);
+  Register res = _result->as_register();
+  ce->store_parameter(_obj->as_register(), 0);
+  if (res != rax) {
+    // This preserves rax and allows it to be used as return-register,
+    // without messing with the stack.
+    __ xchgptr(rax, res);
+  }
```

Comment:
I am trying to wrap my head around this in early Monday :) Is `res` guaranteed to be callee-save? I.e. does this stashing work if runtime call clobbers `res`?

Reply:
OK, taking that one on faith :)